torch/_prims/__init__.py

```diff
@@ -1814,14 +1814,19 @@ def _as_strided_scatter_meta(
     utils.validate_shape(size)
     utils.validate_strides(stride)
 
-    required_size = utils.compute_required_storage_length(size, stride, storage_offset)
+    input_length = utils.compute_required_storage_length(
+        input.size(), input.stride(), input.storage_offset()
+    )
+    required_view_length = utils.compute_required_storage_length(
+        size, stride, storage_offset
+    )
     utils.check(
-        input.numel() >= required_size,
+        input_length >= required_view_length,
```

Comment:
this is not correct, e.g. if I do `torch.as_strided_scatter(inp[1], (1, ), (1,), 2)` this is valid, but your check will fail.

Reply:
You raised a good point, the original code also failed with above test case. 
The behavior I observed in as_strided_scatter, when input is a view like x[1], it makes a copy of view's original tensor x, let's say x', applied src, then return the x'[1]. 

For example:

```
>>> x = torch.ones(4, 1)      // x = [[1],[1],[1],[1]]
>>> src = torch.ones(1)*2
>>> torch.as_strided_scatter(x[1], src, (1,), (1,), 0)  // x' =[[2],[1],[1],[1]], return x'[1] = [1]
tensor([1.])
>>> torch.as_strided_scatter(x[1], src, (1,), (1,), 1)  // x' =[[1],[2],[1],[1]], return x'[1] = [2]
tensor([2.])
>>> torch.as_strided_scatter(x[1], src, (1,), (1,), 2)  // x' =[[1],[1],[2],[1]], return x'[1] = [1]
tensor([1.])
```
If that's the expected behavior, I'm wondering how to check the size properly, as x[1] doesn't know the size of original x. Any suggestion?